vercel/next.js
# Update handling of autoscrolling on navigation in app

- Scrolls to the top of the page if the changed part below the common layout is still in view when at the top of the document
	- This ensures cases where e.g. the header can stay in view when navigating
	- When the element is not in the viewport when at the top of the page a scroll to that element will be triggered in order to ensure the changed part is always in the viewport.
	- Focus is always put on the changed element regardless of scroll position.
- Add support for scrolling down when navigating, there was a bug where the element would not be scrolled into view if it was down from the current viewport.

Tests:
- Add tests for vertical scrolling
- Add tests for horizontal scrolling (not supported)
- Add tests for `router.refresh()`
	- should not scroll by itself
	- should not block router.push from scrolling (when used as part of a transition: `startTransition(() => { router.push('/'); router.refresh(); }`
	- should not scroll when page refreshed using fast refresh

Test tooling:
- Improved typings on `BrowserInterface` a little - backward compatible change.
fix NEXT-181 ([link](https://linear.app/vercel/issue/NEXT-181))
